packages/http/src/lib.ts

```diff
@@ -125,9 +126,16 @@ const libDefinition = {
       },
     },
   },
-} as const;
-
-const httpLib = createTypeSpecLibrary(libDefinition);
-const { reportDiagnostic, createDiagnostic, createStateSymbol } = httpLib;
+  linter: {
+    rules: [operationReferenceContainerRouteRule],
+    ruleSets: {
+      all: {
+        enable: {
+          [`@typespec/http/${operationReferenceContainerRouteRule.name}`]: true,
```

Comment:
I feel like we need to add this rule to the azure-core ruleset in a separate PR

Reply:
Sent PR https://github.com/Azure/typespec-azure/pull/3546 to add this